### PR TITLE
Add example on the California housing dataset illustrating pitfalls of PFI

### PR DESCRIPTION
**Pull request by jpaillard**

### CODE REVIEW

File: examples/plot_pitfalls_permutation_importance.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    mask=mask,
+    cmap="coolwarm",
+    annot=True,
+    fmt=".2f",
+    square=True,
+    cbar_kws={"shrink": 0.8},
+    ax=ax,
+)
+ax.set_title("Correlation Matrix")
+ax.set_yticklabels(feature_names, fontsize=10)
```
Comment (bthirion) [inline examples/plot_pitfalls_permutation_importance.py:65]: could be more readable if the labels were displayed horizontally or in oblique.

**Review by bthirion — COMMENTED** [review]
The example takes ages to run... 
We should profile it a save time.

File: examples/plot_pitfalls_permutation_importance.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+pval_pi = ttest_1samp(
+    permutation_importances, 0.0, axis=0, alternative="greater"
+).pvalue
+
+
+# Define a p-value threshold
+pval_threshold = 0.05
+
+# Create a horizontal boxplot of permutation importances
+fig, ax = plt.subplots(figsize=(10, 6))
```
Comment (bthirion) [inline examples/plot_pitfalls_permutation_importance.py:135]: why do we need such a large figure ?

Comment (jpaillard) [reply examples/plot_pitfalls_permutation_importance.py]: I removed the figsize parameter.

File: examples/plot_pitfalls_permutation_importance.py (hunk 1 above)
Comment (jpaillard) [reply examples/plot_pitfalls_permutation_importance.py]: I added a rotation to the labels

**Comment by jpaillard** [comment]
When I last downloaded the doc, it took a bit more than a minute. What duration should we aim for? 
Still, I reduced the number of samples used in the example, reduced the nb of parameters in the NN, added early stopping, increased the learning rate, and reduced the grid for the RidgeCV in the imputation model.

**Comment by bthirion** [comment]
When I run the example on my box, it takes 3 minutes...

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: examples/plot_pitfalls_permutation_importance.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+# ------------------------------------------------------------------
+# The California housing dataset is a regression dataset with 8 features. We add a
+# spurious feature that is a linear combination of 3 features plus some noise.
+# The spurious feature does not provide any additional information about the target.
+
+dataset = fetch_california_housing()
+X_, y_ = dataset.data, dataset.target
+# only use 2/3 of samples to speed up the example
+X, _, y, _ = train_test_split(X_, y_, test_size=0.6667, random_state=0, shuffle=True)
+rng = np.random.RandomState(0)
```
Comment (lionelkusch) [inline examples/plot_pitfalls_permutation_importance.py:42]: Can you put this line with the import?

File: examples/plot_pitfalls_permutation_importance.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+redundant_coef = rng.choice(np.arange(X.shape[1]), size=(3,), replace=False)
+X_spurious = X[:, redundant_coef].sum(axis=1)
+X_spurious += rng.normal(0, scale=np.std(X_spurious) * 0.5, size=X.shape[0])
+X = np.hstack([X, X_spurious[:, np.newaxis]])
+feature_names = dataset.feature_names + ["Spurious"]
```
Comment (lionelkusch) [inline examples/plot_pitfalls_permutation_importance.py:48]: Can you display a few lines of X?

This is to give a better idea of data you are manipulating.

File: examples/plot_pitfalls_permutation_importance.py (hunk 3 above)
Comment (jpaillard) [reply examples/plot_pitfalls_permutation_importance.py:42]: Well, it's an array of floats, I'm not sure it would be so insightful. However, in the example, [fetch_california_housing](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html#sklearn.datasets.fetch_california_housing) is linked to the description of the dataset in sklearn's doc.

File: examples/plot_pitfalls_permutation_importance.py (hunk 4 above)
Comment (jpaillard) [reply examples/plot_pitfalls_permutation_importance.py:48]: Well, it's an array of floats, I'm not sure it would be so insightful. However, in the example, [fetch_california_housing](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html#sklearn.datasets.fetch_california_housing) is linked to the description of the dataset in sklearn's doc.

Comment (lionelkusch) [reply examples/plot_pitfalls_permutation_importance.py:49]: in this case, you can do just like this example: https://scikit-learn.org/stable/auto_examples/ensemble/plot_forest_hist_grad_boosting_comparison.html#load-dataset
`
print(f"The dataset consists of {n_samples} samples and {n_features} features")`

Comment (jpaillard) [reply examples/plot_pitfalls_permutation_importance.py:49]: done

**Comment by lionelkusch** [comment]
Can you fix this two warnings?
`/home/circleci/project/examples/plot_pitfalls_permutation_importance.py:165: UserWarning: set_ticklabels() should only be used with a fixed number of ticks, i.e. after set_ticks() or using a FixedLocator.
  ax.set_yticklabels(feature_names, fontsize=10)  # Set y-tick labels to feature names`
  (I think you need to transform feature to a list)

`/home/circleci/project/examples/plot_pitfalls_permutation_importance.py:229: SyntaxWarning: invalid escape sequence '\l'  ax.set_xlabel("-$\log_{10}(pval)$")`

After it will be good for me.

**Comment by jpaillard** [comment]
should be good

**Review by lionelkusch — APPROVED** [review]
Good for me
